### ISSUE TITLE
Request for Code on Discrete Latent Priors

### ISSUE THREAD

**Issue by NafiseHaghshenas** [isjakewong/InfoDiffusion#5]
I’m opening a new issue since the previous one was closed, and I wasn't sure if my message was seen.

Apologies for the double post.

Could you please guide me on where I might find the code for the Discrete Latent Priors used in your experiment? If it hasn't been included in the repository yet, would it be possible for you to share it?

Thank you in advance for your help!
Nafise

**Comment by isjakewong** [comment]
Hi Nafise, thanks for your interests in our work. We didn't open-source the discrete latent part but it's easy to implement. You can apply a concrete distribution on the latent using Gumbel-Softmax trick to realize this. You can refer to this repo: https://github.com/YongfeiYan/Gumbel_Softmax_VAE/blob/master/gumbel_softmax_vae.py.

**Comment by NafiseHaghshenas** [comment]
Hi,

First, I'd like to say thank you for your help and suggestion. It has been really useful. 

I have a follow-up question regarding the implementation of Maximum Mean Discrepancy (MMD) loss in the context of categorical distributions.
In the code, I have the following:

a, a_q, mu, log_var = self.encoder(x)

In the case of a categorical distribution, instead of having continuous values like mu (which represents means), we work with the Gumbel-Softmax sampled values and the softmax of the output logits.

Z_gumble_softmax_sample, qy = self.encoder(x)

For the KLD loss in categorical setup, the formula seems to be:

log_ratio = torch.log(qy * categorical_dim + 1e-20)
KLD = torch.sum(qy * log_ratio, dim=-1).mean()

Where qy represents the predicted categorical probabilities.

Now, when calculating the MMD loss, we have:

loss_mmd = compute_mmd(true_samples, mu)

I’m unclear about what true_samples and mu represent in the case of categorical distributions. How should these values be interpreted in this context? Specifically, what should mu represent for categorical distributions, and what exactly are true_samples in this case?

Any insights would be greatly appreciated. Thanks!
Nafise
